modules/gdscript/gdscript_parser.cpp

```diff
@@ -3060,7 +3060,7 @@ void GDScriptParser::_parse_block(BlockNode *p_block, bool p_static) {
 				current_block = p_block;
 				if (error_set)
 					return;
-				p_block->has_return = cf_while->body->has_return;
+				p_block->has_return = false; // while body may never execute
```

Comment:
`has_return` is already `false` by default, you can just remove the line. Resetting it to `false` may be wrong if there was a `return` before the `while` statement.